### PR TITLE
DistributionAggregation fixes

### LINKED ISSUE #373: Multiple apparent bugs in DistributionAggregation implementation
It looks like there were a few bugs added in #149 and #287:

Exemplars are always attached to the last bucket:

```python
from opencensus.stats.aggregation import DistributionAggregation

da = DistributionAggregation(list(map(float, range(0, 11))))
ad = da.aggregation_data

for dp in range(10):
    ad.add_sample(dp, dt.now(), {"trace_id": "dead", "span_id": "beef"})
    
ad.exemplars
```
```python
{11: <opencensus.stats.aggregation_data.Exemplar at 0x10ccf6390>}
```

Off-by-one error adding a value greater than the last bucket boundary:

```python
from datetime import datetime as dt

da = DistributionAggregation([1, 3, 5])
ad = da.aggregation_data

ad.add_sample(float('-inf'), dt.now(), {"trace_id": "dead", "span_id": "beef"})
ad.add_sample(2, dt.now(), {"trace_id": "dead", "span_id": "beef"})
ad.add_sample(4, dt.now(), {"trace_id": "dead", "span_id": "beef"})
ad.add_sample(float('inf'), dt.now(), {"trace_id": "dead", "span_id": "beef"})
  
ad.counts_per_bucket
```
```python
[1, 1, 2, 0]
```

Min/max are 0 by default, which doesn't work for e.g. histograms without 0-valued points:

```python
da = DistributionAggregation(list(map(float, range(0, 11))))
ad = da.aggregation_data

for dp in range(10, 20):
    ad.add_sample(dp, dt.now(), {"trace_id": "dead", "span_id": "beef"})
    
ad.min  # should be 10
```
```python
0
```
...or histograms with all negative points:

```python
da = DistributionAggregation(list(map(float, range(0, 11))))
ad = da.aggregation_data

for dp in range(-20, -10):
    ad.add_sample(dp, dt.now(), {"trace_id": "dead", "span_id": "beef"})
    
ad.max  # should be -10
```
```python
0
```

### PR DESCRIPTION
**Pull request by c24t**
This diff addresses a few bugs in `DistributionAggregation` and `DistributionAggregationData` from #373 that needed to be fixed before converting these aggregations to metrics for #335.

This is a noisy diff since I formatted and fixed old lint errors in the files I touched. I'll comment on the important changes.

Fixes #373.

### CODE REVIEW

**Comment by songy23** [comment]
Please fix the lint.
```bash
opencensus/stats/aggregation_data.py:221:5: E303 too many blank lines (2)
nox > Command flake8 --exclude=opencensus/trace/exporters/gen/opencensus/ opencensus/ failed with exit code 1
nox > Session lint failed. :(
```

File: tests/unit/stats/test_measure_to_view_map.py [hunk 1]
```diff
@@ -229,12 +221,11 @@ def test_record(self):
         view_columns = ["testTag1", "testColumn2"]
         view_measure = measure
         view_aggregation = mock.Mock()
-        view = View(
-            name=view_name,
-            description=view_description,
-            columns=view_columns,
-            measure=view_measure,
-            aggregation=view_aggregation)
+        View(name=view_name,
```
Comment (songy23) [inline tests/unit/stats/test_measure_to_view_map.py]: Why is this created but never used?

File: tests/unit/stats/test_measure_to_view_map.py [hunk 2]
```diff
@@ -343,7 +315,7 @@ def test_record_with_exporter(self):
         view_columns = ["testTag1", "testColumn2"]
         view_measure = measure
         view_aggregation = mock.Mock()
-        view = View(
+        View(
```
Comment (songy23) [inline tests/unit/stats/test_measure_to_view_map.py]: ditto

File: tests/unit/stats/test_measure_to_view_map.py [hunk 3]
```diff
@@ -378,18 +350,13 @@ def test_export(self):
...
```
Comment (songy23) [inline tests/unit/stats/test_measure_to_view_map.py]: ditto

File: opencensus/stats/aggregation_data.py [hunk 4]
```diff
@@ -123,13 +128,14 @@ def __init__(self,
             bounds = []
 
         if counts_per_bucket is None:
-            counts_per_bucket = []
-            bucket_size = len(bounds) + 1
-            for i in range(bucket_size):
-                counts_per_bucket.append(0)
+            counts_per_bucket = [0 for ii in range(len(bounds) + 1)]
+        elif len(counts_per_bucket) != len(bounds) + 1:
+            raise ValueError("counts_per_bucket length does not match bounds "
```
Comment (mayurkale22) [inline opencensus/stats/aggregation_data.py]: I think we can change to ```"counts_per_bucket has X fields, which is more than the Y buckets allowed."``` where, X:len(counts_per_bucket) and Y: len(bounds) + 1. What do you think?

File: tests/unit/stats/test_measure_to_view_map.py (hunk 1 above)
Comment (c24t) [reply tests/unit/stats/test_measure_to_view_map.py]: I didn't start the fire here, I just removed the unused `view` var to fix the [F841](https://lintlyci.github.io/Flake8Rules/rules/F841.html) lint error, and same for the changes below.

Many of the tests in this module seem half-baked, but I think this is a problem for another PR.

File: opencensus/stats/aggregation_data.py (hunk 4 above)
Comment (c24t) [reply opencensus/stats/aggregation_data.py]: One benefit of fixed error messages is that you can grep for the exact text, but I'm happy with either here.

File: tests/unit/stats/test_measure_to_view_map.py (hunk 1 above)
Comment (songy23) [reply tests/unit/stats/test_measure_to_view_map.py]: SGTM

**Pull request merged**